API for Professor Profile

Mari-Hayashi opened the pull request:
Implemented a backend API to create faculty member profile. /facultyMemberProfile/create

sydneypun committed:
Initial version of faculty member profile form

sydneypun committed:
Initial version of Faculty Member profile form

sydneypun committed:
Changed biography arrangement and renamed files and folders

Commit:
Starting on user sign up route

Commit:
First commit

sydneypun committed:
Delete index.tsx

sydneypun committed:
Delete api.ts

sydneypun committed:
Pull request edits.

Commit:
Implemented faculty member create function

Commit:
Implemented faculty member create function

johan1505 commented on src/server/src/modules/facultyMember.ts:
```diff
@@ -0,0 +1,42 @@
+ * @param title string
+ * @returns Promise
+ */
+export const createFacultyMember = (
```
I think this function should be a "updateFacultyMember", meaning it should update the information of an already existing faculty member. This because my 'sign-up' route will be the one responsible for actually creating a faculty member with some initial data(names, email, password). Then, when the faculty member has been registered, he/she can update his/her information by adding the things that you get in this function like biography, department, etc.

Mari-Hayashi replied:
Do you mean we want two APIs?
1. An API that takes in initial data (names, email, and password) and creates a professor profile
2. An API to update the professor profile with the detailed info

If so, do we want to add the "password" field to the facultyMember entity?

johan1505 replied:
Yes. the API that takes initial data and creates a professor record is handled by my authentication API (the authentication task that I'm working on). The API for updating a professor profile is what I think this task should be. I added the password fields to student and facultyMember entities in my user authentication task. I will be done with it by this afternoon and will assign you as a reviewer.

Mari-Hayashi replied:
How do we update "user" field inside "facutlyMember"?

getRepository(FacultyMember).update(id, {
            departmentId,
            websiteLink,
            office,
            title,
            user
        });
makes "user" field of "facultyMember" null.

Mari-Hayashi replied:
Done.

johan1505 commented on src/server/src/routes/facultyMember.ts:
```diff
@@ -0,0 +1,80 @@
+ *   POST Request example - Create - "POST /api/facultyMemberProfile/create"
+ ******************************************************************************/
+
+router.post('/create', async (req: facultyMemberRequest, res: Response) => {
```
create -> update

johan1505 commented on src/server/src/routes/job.ts:
```diff
@@ -59,23 +59,6 @@ router.post('/create', async (req: jobRequest, res: Response) => {
         });
     }
     try {
-        // Check if required field is missing.
```
Why is this being deleted

Mari-Hayashi replied:
There is exactly the same thing a few lines above.

johan1505 replied:
oh lol

johan1505 reviewed (changes requested):
Left comments.

sydneypun committed:
Fixed API file

Commit:
finishes passport set up

Commit:
Adds JWT env variable

Commit:
remvoes unnecessary constructors

Commit:
Updates tsconfig

Commit:
add absolute path for lib folder and add jwt file

Commit:
finises user route

Commit:
finishes helper modules

Commit:
updates entities schemas en relations

Commit:
Coompletes passport set up

Commit:
removes log

sydneypun committed:
Update routes.tsx

sydneypun committed:
Fixed routes.tsx

Commit:
adds confirmed passowrd

Commit:
Adds documentation of user API

Commit:
adds requested change

Commit:
The API now updates user fields.

Mari-Hayashi committed:
Merge branch 'main' into MakeProfAccountBack

Commit:
Removed unnecessary change

Mari-Hayashi committed:
Update routes.tsx

Mari-Hayashi committed:
Update routes.tsx

Commit:
Updates appropiate fields for facultymember and user records

johan1505 reviewed (commented):
I added a commit that makes the API update the user and faculty member records. I also removed some parameters such as password or email that I don't think should be updated but this API as they require extra validation. Let me know if the changes make sense.

Commit:
Updated readme

Mari-Hayashi commented:
> I added a commit that makes the API update the user and faculty member records. I also removed some parameters such as password or email that I don't think should be updated but this API as they require extra validation. Let me know if the changes make sense.

I updated the readme and tested the program. It works fine. Please let me know if there is anything I can do.
